WordPress/WordPressTest/CoreDataStackHelper.swift

```diff
@@ -0,0 +1,49 @@
+import Combine
+import XCTest
+
+@testable import WordPress
+
+extension ContextManagerMock {
```

Comment:
I'm not sure it's worth creating a new, dedicated `CoreDataStackHelper.swift` file here:
1. The file naming does not respect the conventions of matching the class/type/extension it defines, so when navigating the files list, it makes it harder to understand just from the name what it might contain. You're not defining a new `class CoreDataStackHelper` here. If we were to keep a dedicated file for it, I'd rather go with something like `ContextManagerMock+TestHelper.swift` or something similar
2. There are already a lot of different files and classes around `CoreDataStack` and its test helpers and mocks, so adding a new file/name/type doesn't help in simplifying stuff and getting an easier-to-grasp picture imho
3. Given my suggestion below that will simplify this whole file in just a single method with 2 lines of code, at that point I feel it makes more sense to declare the `useAsSharedInstance(untilTestFinished:)` method directly in the same file as where the `ContextManagerMock` is originally defined.

Reply:
I've renamed this file to `ContextManagerMock+AutomaticTeardown.swift`. But the implementation was kept as category in this Swift file, rather than moving to the definition Objective-C files, since the `ContextManagerMock` will become super simple in the follow up PR of https://github.com/wordpress-mobile/WordPress-iOS/pull/18561. I'm thinking I may as well translate `ContextManagerMock` to Swift in that follow up PR, at which point this extension and the class definition will be in the same file.

Reply:
BTW, https://github.com/wordpress-mobile/WordPress-iOS/pull/18630 is the follow up PR I mentioned above.

Reply:
I've [rewritten ContextManagerMock in Swift](https://github.com/crazytonyli/WordPress-iOS/commit/854c7f93e2eb2a4ba0c42ef34cccf72c10c348e8), putting the class definition and the method added in this PR into one file. But, as a result of it, I needed to [add whole lots](https://github.com/crazytonyli/WordPress-iOS/compare/automatic-tear-down...context-manager-mock-swift) of `import XCTest`, since XCTest was imported into the bridge header file [via ContextManagerMock.h](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/WordPressTest/ContextManagerMock.h#L2). I don't want to pollute this PR with this unrelated changes, I think I'm going to leave this PR as it is, and open a new one for the Swift rewrite, do you think that'll work?

Reply:
Ah, your comment and my review ran into a race condition 😂  Didn't see your reply before I hit submit on my review and [mentioned that I'm ok with the rewrite being in separate PR here](https://github.com/wordpress-mobile/WordPress-iOS/pull/18579#discussion_r875747682) 🙂

Reply:
Here is the rewrite PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/18655